src/Interpreter/Interpreter.cpp

```diff
@@ -64,6 +67,13 @@ void Interpreter::execute() {
   traceblock.close();
 }
 
+void Interpreter::visit(llvm::Instruction& inst) {
+  if (interp->caffeine().options().run_line_coverage) {
+    getInstLine(inst);
+  }
```

Comment:
Instead of an option here you could also check whether the context has an instance of the coverage tracker
```suggestion
  if (interp->caffeine().coverage()) {
    getInstLine(inst);
  }
```